content/master/guides/composition-functions.md

```diff
@@ -0,0 +1,1002 @@
+  name: example
+spec:
+  compositeTypeRef:
+    apiVersion: database.example.org/v1alpha1
+    kind: XPostgreSQLInstance
+  functions:
+  - name: my-cool-Function
+    # Currently only Container is supported. Other types may be added in future.
+    type: Container
+    # Configuration specific to type: Container.
+    container:
+      # The OCI image to pull and run.
+      image: xkpg.io/my-cool-Function:0.1.0
+      # Whether to pull the Function image Never, Always, or IfNotPresent.
+      imagePullPolicy: IfNotPresent
+      # Note that only resource limits are supported - not requests.
+      # The Function will be run with the specified resource limits, specified
+      # in Kubernetes-style resource.Quantity form.
+      resources:
+        limits:
+          # Defaults to 128Mi
+          memory: 64Mi
+          # Defaults to 100m (a 10th of a core)
+          cpu: 250m
+      # Defaults to 'Isolated' - i.e an isolated network namespace with no
+      # network access. Use 'Runner' to allow a Function access to the runner's
+      # (e.g. the xfn container's) network namespace.
+      network:
+        policy: Runner
+      # How long the Function may run before it's killed. Defaults to 20s.
+      # Keep in mind the Function pipeline is typically invoked once every
+      # 30 to 60 seconds.
+      timeout: 30s
+    # An arbitrary Kubernetes resource. Passed to the Function as the config
+    # block of its FunctionIO. Doesn't need to exist as a Custom Resource (CR),
+    # since this resource doesn't exist by itself in the API server but must be
+    # a valid Kubernetes resource (i.e. have an apiVersion and kind).
+    config:
+      apiVersion: database.example.org/v1alpha1
+      kind: Config
+      metadata:
+        name: cloudsql
+      spec:
+        version: POSTGRES_9_6
+```
+{{< /expand >}}
+
+Use `kubectl describe <xr-kind> <xr-name>` to debug Composition Functions. Look
+for status conditions and events. Most Functions will emit events associated
+with the XR if they experience issues.
+
+## Building a Function
+
+ Crossplane is relatively unopinionated about how a Composition Function is
+ implemented. Functions must:
+
+ * Be packaged as an OCI image, where the `ENTRYPOINT` is the Function.
+ * Accept input in the form of a `FunctionIO` document on stdin.
+ * Return the `FunctionIO` they were passed, optionally mutated, on stdout.
+ * Run within the constraints specified by the Composition that includes them
+   (e.g. timeouts, compute, network access).
+
+This means Functions may be written using a general purpose programming language
+like Python, Go, or TypeScript. They may also be implemented using a shell
+script, or an existing tool like Helm or Kustomize.
+
+### FunctionIO
+
+When a Composition Function runner like `xfn` runs your Function it will write
+`FunctionIO` to its stdin. A `FunctionIO` is a Kubernetes style YAML manifest.
+It's not a custom resource (it never gets created in the API server) but it does
```

Comment:
Why remove this word? Are you going for "but it follows Kubernetes conventions" rather than "but it does follow Kubernetes conventions"? If so, can you clarify why?

Reply:
Yes, the line becomes "it follows Kubernetes conventions", which is shorter and simpler. "Does" is unnecessary. 

That would be fine for more prose-style writing but is less useful for technical documentation that intends to be straight forward without additional color.